### PR TITLE
Add check for missing result .json files

### PR DESCRIPTION
**Pull request by hannahbaumann**
## Licensing agreement
* [x] I agree to release these inputs under the combined MIT and CCBY SA 4.0 licenses of this repository

[templates]: https://github.com/OpenFreeEnergy/IndustryBenchmarks2024/tree/main/industry_benchmarks
[input_validation]: https://github.com/OpenFreeEnergy/IndustryBenchmarks2024/tree/main/industry_benchmarks/utils/input_validation.py

### CODE REVIEW

File: industry_benchmarks/utils/extras/extract_results.py [hunk 1]
```diff
@@ -138,42 +188,105 @@ def extract(results_0, results_1, results_2, output):
...
+                missing_files.append(f"solvent_{molA}_{molB} repeat {i}")
+            if (molA, molB) not in transform_dict["complex"]:
+                missing_files.append(f"complex_{molA}_{molB} repeat {i}")
+
+    # Before starting to raise errors, print out a summary of health checks
+    click.echo("=" * 80)
+    click.echo("Summary of checks for missing files and file with errors")
+    click.echo("=" * 80)
+    click.echo(
+        "Total number of transformations (both solvent and complex): "
```
Comment (hannahbaumann) [inline industry_benchmarks/utils/extras/extract_results.py:227]:
```suggestion
        "Total number of transformations that should have completed (both solvent and complex): "
```

File: industry_benchmarks/utils/extras/extract_results.py [hunk 2]
```diff
@@ -138,42 +188,105 @@ def extract(results_0, results_1, results_2, output):
...
+            f"{len(files_with_errors) + len(missing_files)}"
+        )
+        click.echo(
+            "    - Number of transformations with errors in the result "
+            f".json file: {len(files_with_errors)}"
+        )
+        click.echo(
+            "    - Number of transformations that did not output a "
+            f".json file: {len(missing_files)}"
+        )
```
Comment (IAlibay) [inline industry_benchmarks/utils/extras/extract_results.py:246]:
```suggestion
            "    - Number of transformations without output "
            f"results files: {len(missing_files)}"
        )
```
Something like that

File: industry_benchmarks/utils/extras/extract_results.py [hunk 3]
```diff
@@ -138,42 +188,105 @@ def extract(results_0, results_1, results_2, output):
...
+            if (molA, molB) not in transform_dict["complex"]:
+                missing_files.append(f"complex_{molA}_{molB} repeat {i}")
+
+    # Before starting to raise errors, print out a summary of health checks
+    click.echo("=" * 80)
+    click.echo("Summary of checks for missing files and file with errors")
+    click.echo("=" * 80)
+    click.echo(
+        "Total number of transformations that should have completed (both solvent and complex): "
+        f"{len(input_ligand_network.edges)*2}"
```
Comment (IAlibay) [inline industry_benchmarks/utils/extras/extract_results.py:228]: A reminder to re-visit this once we look at "fixed" networks.

**Review by IAlibay — CHANGES_REQUESTED** [review]
Couple of things, mostly removing an errand method and clarifying a message. Otherwise lgtm.

**Review by IAlibay — CHANGES_REQUESTED** [review]
Couple of things, mainly removing an extra method and clarifying an error message.

**Review by IAlibay — APPROVED** [review]
lgtm!

**Pull request merged**